apps/script/convert-scss-variables.js

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env node
+
+/**
+ * Script that processes an scss file containing a set of variables, and
+ * generates the equivalent set in JS.
+ */
+var fs = require('fs');
+var path = require('path');
+var readline = require('readline');
+
+// Regular expression to capture a color definition from SCSS
+var variableRe = /\$([\w-]+)\s*:\s*([^;]+);/;
+var cachedVariables = {};
+
+// In color.scss some color definitions reference previous
+// color definitions.  Since we're reading them in order,
+// we can recursively resolve these referential defintions
+// until we land on an actual color.
+function resolveVariable(value) {
+  var originalValue = value;
+  while (/^\$/.test(value)) {
+    value = cachedVariables[value.substring(1)];
+  }
+  return value;
+}
+
+function convertScssToJs(scssPath, jsPath) {
+  // Generate <foo>.js while reading <foo>.scss line-by-line
+  var out = fs.createWriteStream(jsPath);
+  out.write([
+    '// color.js',
+    '// GENERATED FILE: DO NOT MODIFY DIRECTLY',
+    '// This generated file exports all colors defined in ' + scssPath,
+    '// for use in JavaScript. The generator script is convert-scss-variables.js',
+    'module.exports = {\n'
+  ].join('\n'));
+
+  var rl = readline.createInterface({
+    input: fs.createReadStream(scssPath),
+    terminal: false
+  });
+
+  var currentLine = 0;
+  rl.on('line', function (line) {
+    currentLine++;
+    var match = variableRe.exec(line);
+    if (match === null) {
+      return;
+    }
+
+    var variableName = match[1];
+    var variableValue = resolveVariable(match[2]);
+    if (typeof variableValue === 'undefined') {
+      throw new Error([
+          'Unable to resolve color ' + variableName,
+          scssPath + ':' + currentLine,
+          line,
+          ' ^'
+          ].join('\n'));
+    }
+    // JS doesn't play as nicely with with dashes, so replace with underscores
+    variableName = variableName.replace(/-/g, '_');
+    var isNumber = false;
+    if (/px$/.test(variableValue)) {
+      variableValue = parseFloat(variableValue);
+    }
+    cachedVariables[variableName] = variableValue;
+
+    // Put quotes around non-numbers
+    if (!isNumber) {
+      variableValue = '"' + variableValue + '"';
+    }
```

Comment:
You could generalize a little by changing your isNumber regex to something like this:
```javascript
/^[\d\.]+(?:px)?$/
```
But for the most part I think we can do what works now, and expect to adapt down the road.